cartographer/mapping/internal/3d/pose_graph_3d.cc

```diff
@@ -111,31 +111,33 @@ NodeId PoseGraph3D::AddNode(
   const transform::Rigid3d optimized_pose(
       GetLocalToGlobalTransform(trajectory_id) * constant_data->local_pose);
 
-  common::MutexLocker locker(&mutex_);
-  AddTrajectoryIfNeeded(trajectory_id);
-  if (!CanAddWorkItemModifying(trajectory_id)) {
-    LOG(WARNING) << "AddNode was called for finished or deleted trajectory.";
-  }
-  const NodeId node_id = data_.trajectory_nodes.Append(
-      trajectory_id, TrajectoryNode{constant_data, optimized_pose});
-  ++data_.num_trajectory_nodes;
-
-  // Test if the 'insertion_submap.back()' is one we never saw before.
-  if (data_.submap_data.SizeOfTrajectoryOrZero(trajectory_id) == 0 ||
-      std::prev(data_.submap_data.EndOfTrajectory(trajectory_id))
-              ->data.submap != insertion_submaps.back()) {
-    // We grow 'data_.submap_data' as needed. This code assumes that the first
-    // time we see a new submap is as 'insertion_submaps.back()'.
-    const SubmapId submap_id =
-        data_.submap_data.Append(trajectory_id, InternalSubmapData());
-    data_.submap_data.at(submap_id).submap = insertion_submaps.back();
-    LOG(INFO) << "Inserted submap " << submap_id << ".";
+  NodeId node_id(0, 0);
```

Comment:
The constructor for `NodeId` was introduced to prevent uninitialized IDs. This seems not much better. How about moving the code under the mutex lock into a separate function? Then you can take the mutex there and return the `node_id`.

Reply:
Done.